pyccl/tests/test_hashing.py

```diff
@@ -0,0 +1,34 @@
+"""Test the hashing function of CCL."""
+import pytest
+import pyccl as ccl
+import numpy as np
+from collections import OrderedDict
+
+OBJECTS = [ccl.Cosmology,  # class
+           (0, 1, 2),      # tuple
+           [0, 1, 2],      # list
+           set([0, 1, 2]),   # set
+           np.arange(3),   # array
+           {0: None, 1: None, 2: None},                    # dict
+           {0: None, 1: None, 2: {2.1: None, 2.2: None}},  # nested dict
+           OrderedDict({0: None, 1: None, 2: None}),       # OrderedDict
+           ccl.CosmologyVanillaLCDM(),  # something else
```

Comment:
humm, can you walk me through how this hashes a cosmology?

Reply:
Sure, so it falls under the "something else" category of the `hash_` function, i.e. it can't be reliably hashed by that function. So, it simply calls the `__hash__` method of this object:
```
    elif hasattr(obj, "__hash__"):
        # Hashables: Just return the object.
        return obj
```

The idea is that we use the representation of the object for hashing, since, representations should strictly be unique, as per Python guidelines. And we simply hash the representation string.

Reply:
Now, in the `CCLObject` PR, I've created a module `_repr.py` which contains repr methods for all objects in CCL. You can have a look there to see specifically how I'm thinking of implementing hashing for `Cosmology` objects, but this is up to discussion.